GP/toy_experiments/toy_sin_noise.py

```diff
@@ -148,111 +150,50 @@
+    rand_errs = np.sqrt(rand_squares / random_trials - rand_means ** 2)
     homo_errs = np.sqrt(homo_squares / random_trials - homo_means ** 2)
     hetero_errs = np.sqrt(hetero_squares / random_trials - hetero_means ** 2)
-    aug_means = aug_running_sum / random_trials
-    aug_errs = np.sqrt(aug_squares / random_trials - aug_means ** 2)
-    aug_het_means = aug_het_running_sum / random_trials
-    aug_het_errs = np.sqrt(aug_het_squares / random_trials - aug_het_means **2)
 
+    print('List of average random sampling values is: ' + str(rand_means))
+    print('List of random sampling errors is:' + str(rand_errs))
     print('List of average homoscedastic values is: ' + str(homo_means))
     print('List of homoscedastic errors is: ' + str(homo_errs))
     print('List of average heteroscedastic values is ' + str(hetero_means))
     print('List of heteroscedastic errors is: ' + str(hetero_errs))
-    print('List of average AEI values is: ' + str(aug_means))
-    print('List of AEI errors is: ' + str(aug_errs))
-    print('List of average het-AEI values is: ' + str(aug_het_means))
-    print('List of het-AEI errors is: ' + str(aug_het_errs))
 
     iter_x = np.arange(1, bayes_opt_iters + 1)
+    iter_rand_x = np.arange(1, random_sampling_iters + 1)
+
+    # clear figure from previous fplot returns if fiddling with form of function
+    plt.cla()
 
     ax = plt.gca()
     ax.xaxis.set_major_locator(MaxNLocator(integer=True))
     plt.plot(iter_x, homo_means, color='r', label='Homoscedastic')
     plt.plot(iter_x, hetero_means, color='b', label='Heteroscedastic ANPEI')
-    #plt.plot(iter_x, aug_means, color='g', label='Homoscedastic AEI')
-    plt.plot(iter_x, aug_het_means, color='c', label='Heteroscedastic AEI')
+    plt.plot(iter_rand_x, rand_means, color='g', label='Random Sampling')
+    lower_rand = np.array(rand_means) - np.array(rand_errs)
+    upper_rand = np.array(rand_means) + np.array(rand_errs)
     lower_homo = np.array(homo_means) - np.array(homo_errs)
     upper_homo = np.array(homo_means) + np.array(homo_errs)
     lower_hetero = np.array(hetero_means) - np.array(hetero_errs)
     upper_hetero = np.array(hetero_means) + np.array(hetero_errs)
-    lower_aei = np.array(aug_means) - np.array(aug_errs)
-    upper_aei = np.array(aug_means) + np.array(aug_errs)
-    lower_het_aei = np.array(aug_het_means) - np.array(aug_het_errs)
-    upper_het_aei = np.array(aug_het_means) + np.array(aug_het_errs)
     plt.fill_between(iter_x, lower_homo, upper_homo, color='r', label='Homoscedastic', alpha=0.1)
     plt.fill_between(iter_x, lower_hetero, upper_hetero, color='b', label='Heteroscedastic ANPEI', alpha=0.1)
-    #plt.fill_between(iter_x, lower_aei, upper_aei, color='g', label='Homoscedastic AEI', alpha=0.1)
-    plt.fill_between(iter_x, lower_het_aei, upper_het_aei, color='c', label='Heteroscedastic AEI', alpha=0.1)
+    plt.fill_between(iter_rand_x, lower_rand, upper_rand, color='g', label='Random Sampling', alpha=0.1)
```

Comment:
We can use plt.errorbar here together with some combination of label='Heteroscedastic ANPEI' etc to plot errorbars instead of the shaded regions

Reply:
just having a bit of trouble with this as the plt.errorbar function takes no negative values and we have just the magnitudes (i.e. lower_homo, upper_homo) which dip into the negatives in the earlier iterations

Reply:
working on it now though